packages/matchbox/src/hooks/usePrefersColorScheme.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { getWindow } from '../helpers/window';
+
+/**
+ * SSR friendly hook that returns prefers-color-scheme status
+ *
+ * @returns
+ * 'dark' OR 'light'
+ *
+ * @example
+ * const prefersColorScheme = usePrefersColorScheme();
+ * return <div>{prefersColorScheme}</div>
+ */
+
+const QUERY = '(prefers-color-scheme: dark)';
+
+function usePrefersColorScheme() {
+  const environment = getWindow();
+
+  const [prefersColorScheme, setPrefersColorScheme] = React.useState(
+    environment.matchMedia(QUERY).matches,
+  );
+
+  React.useEffect(() => {
+    const mql = environment.matchMedia(QUERY);
+
+    const listener = event => {
+      setPrefersColorScheme(event.matches);
+    };
+
+    mql.addListener(listener);
```

Comment:
The code I posted above doesn't work unless we check for the existence of `addEventListener`

So like:
```js
if (mql.addEventListener){
  mql.addEventListener(listener)
}
```

I guess we'll need the same below for the remove event